rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParserOnlyContribution.java

```diff
@@ -0,0 +1,254 @@
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.rascalmpl.exceptions.RuntimeExceptionFactory;
+import org.rascalmpl.interpreter.Configuration;
+import org.rascalmpl.interpreter.Evaluator;
+import org.rascalmpl.interpreter.asserts.Ambiguous;
+import org.rascalmpl.interpreter.staticErrors.UndeclaredNonTerminal;
+import org.rascalmpl.interpreter.utils.JavaBridge;
+import org.rascalmpl.parser.gtd.IGTD;
+import org.rascalmpl.parser.gtd.exception.ParseError;
+import org.rascalmpl.parser.gtd.exception.UndeclaredNonTerminalException;
+import org.rascalmpl.parser.gtd.recovery.IRecoverer;
+import org.rascalmpl.parser.gtd.result.out.DefaultNodeFlattener;
+import org.rascalmpl.parser.uptr.UPTRNodeFactory;
+import org.rascalmpl.parser.uptr.action.NoActionExecutor;
+import org.rascalmpl.uri.URIResolverRegistry;
+import org.rascalmpl.values.IRascalValueFactory;
+import org.rascalmpl.values.parsetrees.ITree;
+import org.rascalmpl.values.parsetrees.SymbolAdapter;
+import org.rascalmpl.values.parsetrees.TreeAdapter;
+import org.rascalmpl.vscode.lsp.parametric.model.ParametricSummaryBridge;
+import org.rascalmpl.vscode.lsp.terminal.ITerminalIDEServer.ParserSpecification;
+import org.rascalmpl.vscode.lsp.util.concurrent.InterruptibleFuture;
+import io.usethesource.vallang.IConstructor;
+import io.usethesource.vallang.IList;
+import io.usethesource.vallang.ISet;
+import io.usethesource.vallang.ISourceLocation;
+import io.usethesource.vallang.IValue;
+import io.usethesource.vallang.IValueFactory;
+
+public class ParserOnlyContribution implements ILanguageContributions {
+
+    private final String name;
+    private final String extension;
+    private final @Nullable Exception loadingParserError;
+    private final @Nullable Class<IGTD<IConstructor, ITree, ISourceLocation>> parserClass;
+    private final String parserMethodName;
+    private final boolean allowAmbiguity;
+
+    public ParserOnlyContribution(String name, String extension, ParserSpecification spec) {
+        this.name = name;
+        this.extension = extension;
+        Class<IGTD<IConstructor, ITree, ISourceLocation>> clzz = null;
+        Exception err = null;
+        try {
+            clzz = loadParserClass(spec);
+        } catch (ClassNotFoundException | IOException e) {
+            err = e;
+        }
+        this.parserClass = clzz;
+        this.loadingParserError = err;
+        this.parserMethodName = (spec.getNonTerminalIsStart() ? "start__" : "") + spec.getNonTerminalName();
+        this.allowAmbiguity = spec.getAllowAmbiguity();
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getExtension() {
+        return extension;
+    }
+
+    @Override
+    public CompletableFuture<ITree> parseSourceFile(ISourceLocation loc, String input) {
```

Comment:
all of this is clones and can be reused from the above IFunction acquired from `loadParsers`

Reply:
true, but I had to, as to get that `IFunction`, you have to have a Evaluator instance.

Reply:
specifically the `RascalFunctionValueFactory` needs a `IEvaluatorContext`. 

but I'm not a fan of the amount of copying I had to do. completly open for an alternative idea.

Reply:
I rewrote the code to use a `RascalFunctionValueFactory` with an empty `Evaluator` instance. The 4 additional object allocations are not costly, even if they are somewhat ugly. The good news is that we don't leak the parser storage implementation internals and the parser implementation technology is behind the `IRascalValueFactory` interface again. 

When this stuff changes internally in the future, the current code may remain almost the same. When we move to a compiled version of Rascal code, the specific instance of the RascalFunctionValueFactory will have to change, but not the code that calls into IRascalFunctionValueFactory API.

Reply:
@DavyLandman if you have time to test this new code, that would mean a lot. I don't have time to setup a test environment for this one.